### PR TITLE
feat: add code cov on github actions

### PR DESCRIPTION
**Pull request by baby230211**
## Types of changes

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
move code cov to github actions

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -60,9 +60,8 @@ jobs:
     - name: Run Test and Codecov
       run: |
         cd src
-        pytest -n 2 --cov=.
-        codecov
+        pytest -n 2 --cov=./ --cov-report=xml
 
     # TODO: Use Official Codecov from Github Actions
```
Comment (josix) [inline .github/workflows/ci.yaml:65]: **nitpick:** Need to remove this.

**Review by josix — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (baby230211) [reply .github/workflows/ci.yaml]: fixed ~ thanks

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -60,9 +60,7 @@ jobs:
     - name: Run Test and Codecov
       run: |
         cd src
-        pytest -n 2 --cov=.
-        codecov
+        pytest -n 2 --cov=. --cov-report=xml
```
Comment (mattwang44) [inline .github/workflows/ci.yaml:63]: 這樣只會有 report file 但 terminal 上就不會有，但希望之後打開 github action log 還是能看的到 report

Comment (baby230211) [reply .github/workflows/ci.yaml]: updated~

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -60,9 +60,7 @@ jobs:
     - name: Run Test and Codecov
```
Comment (mattwang44) [inline .github/workflows/ci.yaml:60]:
```suggestion
    - name: Run Test
```

**Review by mattwang44 — APPROVED** [review]
Nice job! Only one last minor issue left.

**Review by josix — APPROVED** [review]

**Pull request merged**